openmdao/vectors/petsc_vector.py

```diff
@@ -107,9 +115,16 @@ def _initialize_data(self, root_vector):
         super(PETScVector, self)._initialize_data(root_vector)
 
         self._petsc = {}
+        self._imag_petsc = {}
```

Comment:
It doesn't work collectively, but the we just have to set the `.array` attribute on every processor. I'm certain this works though; I did this in MAUD when looping over columns of the PETSc 'multi-vectors', assigning each column to the `.array` attribute. I'm fine with it if you want to add a story and get this pull request in as is